sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/bitwiseExpressions.scala

```diff
@@ -204,3 +205,60 @@ case class BitwiseCount(child: Expression)
     case LongType => java.lang.Long.bitCount(input.asInstanceOf[Long])
   }
 }
+
+object BitwiseGetUtil {
+  def calculate(target: Any, pos: Int, size: Int): Byte = {
```

Comment:
this is inefficient because of boxing. I'd suggest the following
```
def checkPosition(pos: Int, size: Int) = ...
...
override def nullSafeEval(target: Any, pos: Any): Any = {
  val posInt = pos.asInstanceOf[Int]
  BitwiseGetUtil.checkPosition(posInt, bitSize)
  ((target.asInstanceOf[Number].longValue >> posInt) & 1).toByte
}
override def doGenCode... {
  ...
  BitwiseGetUtil.checkPosition($pos, $bitSize)
  (byte)((((long) target) >> $pos) & 1)
}
```